src/system/AdminModule/Helper/UpdateCheckHelper.php

```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    /**
+     * Zikula curl
+     *
+     * This function is internal for the time being and may be extended to be a proper library
+     * or find an alternative solution later.
+     *
+     * @param string $url
+     * @param int $timeout
+     *            default=5
+     *
+     * @return string|bool false if no url handling functions are present or url string
+     */
+    private function zcurl($url, $timeout = 5)
+    {
+        $urlArray = parse_url($url);
+        $data = '';
+        $userAgent = 'Zikula/' . $this->currentVersion;
+        $ref = $this->requestStack
+                ->getMasterRequest()
+                ->getBaseURL();
+        $port = (($urlArray['scheme'] == 'https') ? 443 : 80);
+        if (ini_get('allow_url_fopen')) {
+            // handle SSL connections
+            $path_query = (isset($urlArray['query']) ? $urlArray['path'] . $urlArray['query'] : $urlArray['path']);
+            $host = ($port == 443 ? "ssl://$urlArray[host]" : $urlArray['host']);
+            $fp = @fsockopen($host, $port, $errno, $errstr, $timeout);
+            if (!$fp) {
+                return false;
+            } else {
+                $out = "GET $path_query? HTTP/1.1\r\n";
+                $out .= "User-Agent: $userAgent\r\n";
+                $out .= "Referer: $ref\r\n";
+                $out .= "Host: $urlArray[host]\r\n";
+                $out .= "Connection: Close\r\n\r\n";
+                fwrite($fp, $out);
+                while (!feof($fp)) {
+                    $data .= fgets($fp, 1024);
+                }
+                fclose($fp);
+                $dataArray = explode("\r\n\r\n", $data);
+
+                return $dataArray[1];
+            }
+        } elseif (function_exists('curl_init')) {
+            $ch = curl_init();
+            curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
+            curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
+            curl_setopt($ch, CURLOPT_URL, "$url?");
+            curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 1);
+            curl_setopt($ch, CURLOPT_USERAGENT, $userAgent);
+            curl_setopt($ch, CURLOPT_REFERER, $ref);
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+            if (!ini_get('safe_mode') && !ini_get('open_basedir')) {
+                // This option doesnt work in safe_mode or with open_basedir set in php.ini
+                curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+            }
+            curl_setopt($ch, CURLOPT_TIMEOUT, $timeout);
+            $data = curl_exec($ch);
+            if (!$data && $port = 443) {
+                // retry non ssl
+                $url = str_replace('https://', 'http://', $url);
+                curl_setopt($ch, CURLOPT_URL, "$url?");
+                $data = @curl_exec($ch);
+            }
+            //$headers = curl_getinfo($ch);
+            curl_close($ch);
+
+            return $data;
+        } else {
+            return false;
+        }
+    }
```

Comment:
Suggestions:

1. use curl primarily
2. Replace the else statements by dedicated conditions, since the single blocks end with return statements